### PR TITLE
Deal with eventual consistency in mysql database API calls

### LINKED ISSUE #270: Creating a new MySQL database - fetching user fails
Creating a new MySQL database works, but fetching the created user fails:

```
mw database mysql create --description="my description" --version="8.0"

  Creating a new MySQL database

  ❓ enter password for default user: ******************************************************************************************************************************** 
  ✅ creating MySQL database. done
  ✅ fetching database. done
  ❌ fetching user. error

     An error occurred during this operation:
     ApiClientError: Unexpected response status (expected 200, got: 403)

╭──────────────────────────────────────────────────────────────────────────────╮
│ API CLIENT ERROR                                                             │
│                                                                              │
│ An error occurred while communicating with the API: Unexpected response      │
│ status (expected 200, got: 403)                                              │
│                                                                              │
│ {                                                                            │
│   "params": {                                                                │
│     "traceId": "f727d21f1f6ad105"                                            │
│   },                                                                         │
│   "message": "access denied; verdict: abstain",                              │
│   "type": "PermissionDenied"                                                 │
│ }                                                                            │
╰──────────────────────────────────────────────────────────────────────────────╯

  ERROR STACK TRACE

  Please provide this when opening a bug report.

  ApiClientError: Unexpected response status (expected 200, got: 403)
      at ApiClientError.fromResponse (file:///var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/node_modules/@mittwald/api-client-commons/dist/esm/core/ApiClientError.js:9:16)
      at assertOneOfStatus (file:///var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/node_modules/@mittwald/api-client-commons/dist/esm/types/assertOneOfStatus.js:4:30)
      at assertStatus (file:///var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/node_modules/@mittwald/api-client-commons/dist/esm/types/assertStatus.js:3:5)
      at file:///var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/dist/commands/database/mysql/create.js:86:13
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async FancyProcessRenderer.runStep (file:///var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/dist/rendering/process/process_fancy.js:42:28)
      at async Create.exec (file:///var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/dist/commands/database/mysql/create.js:82:22)
      at async Create.run (file:///var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/dist/rendering/react/ExecRenderBaseCommand.js:18:24)
      at async Create._run (/var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/node_modules/@oclif/core/lib/command.js:304:22)
      at async Config.runCommand (/var/lib/nvm/versions/node/v18.17.0/lib/node_modules/@mittwald/cli/node_modules/@oclif/core/lib/config/config.js:426:25)

```

### PR DESCRIPTION
**Pull request by martin-helmich**
This PR changes the `database mysql create` command to both respect the documented [Eventual Consistency mechanic](https://developer.mittwald.de/docs/v2/api/intro/#eventual-consistency), and also adds a retry mechanism as a secondary precaution.

Fixes #270

### CODE REVIEW

File: src/commands/database/mysql/create.tsx [hunk 1]
```diff
@@ -71,59 +76,85 @@ export class Create extends ExecRenderBaseCommand<typeof Create, Result> {
...
+    p: ProcessRenderer,
+    projectId: string,
+    description: string,
+    version: string,
+    collation: string,
+    characterSet: string,
+    password: string,
+    externalAccess: boolean,
+    accessLevel: string,
+  ) {
```
Comment (LukasFritzeDev) [inline src/commands/database/mysql/create.tsx:130]: This are quite a lot function parameters. Would be better to use objects o group some of them? i.e. 

```suggestion
  private async createMySQLDatabase(
    p: ProcessRenderer,
    projectId: string,
    dbOptions: {...},
    userOptions: {...},
  ) {
```

File: src/lib/api_retry.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+function isStatus<T extends Response, S extends T["status"]>(
+  response: T,
+  expectedStatus: S,
+): response is T & {
+  status: S;
+} {
+  return response.status === expectedStatus;
+}
```
Comment (LukasFritzeDev) [inline src/lib/api_retry.ts:77]: This function already exists and is in use in this project:

```
import { assertStatus } from "@mittwald/api-client-commons";
```

**Review by LukasFritzeDev — DISMISSED** [review]

File: src/lib/api_retry.ts (hunk 2 above)
Comment (martin-helmich) [reply src/lib/api_retry.ts:77]: Yes, and no... `assertStatus` is an assertion function[^1] that throws an error, whereas `isStatus` is a type predicate [^2], which just _returns_.

[^1]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions
[^2]: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

File: src/commands/database/mysql/create.tsx (hunk 1 above)
Comment (martin-helmich) [reply src/commands/database/mysql/create.tsx]: Good suggestion; I refactored this a bit.

File: src/lib/api_retry.ts (hunk 2 above)
Comment (LukasFritzeDev) [reply src/lib/api_retry.ts:77]: Oh. Sorry. My Bad.

File: src/lib/api_consistency.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    if (lastEventId !== undefined) {
+      d("setting if-event-reached to %o", lastEventId);
+      config.headers["if-event-reached"] = lastEventId;
+    }
+    return config;
+  });
+
+  axios.interceptors.response.use((response) => {
+    const isMutatingRequest =
+      ["post", "put", "delete"].indexOf(
```
Comment (mfal) [inline src/lib/api_consistency.ts:19]:
```suggestion
      ["post", "put", "delete", "patch"].indexOf(
```

File: src/lib/api_retry.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+export function configureAxiosRetry(axios: AxiosInstance) {
+  axios.interceptors.request.use((config) => {
+    return {
+      ...config,
+      validateStatus: (status) => status < 300,
+    };
+  });
+
+  axiosRetry(axios, {
+    retries: 50,
```
Comment (mfal) [inline src/lib/api_retry.ts:16]: Mir erscheint `50` etwas viel.

Comment (martin-helmich) [reply src/lib/api_retry.ts:16]:
```suggestion
    retries: 10,
```

Ich setze es mal auf 10 runter-- wir brauchen auf jeden Fall mehr als die standardmäßigen 3 Retries, da manche Routen tatsächlich _echt_ lange brauchen, bis sich die ACLs aktualisiert haben (unabhängig davon, dass wir das Issue intern natürlich sowieso fixen sollten).

**Review by mfal — APPROVED** [review]

**Pull request merged**